### PR TITLE
create spaceship and move around

### PR DESCRIPTION
**Pull request by chiptus**
This pull requests allows the ship to move up and right although in the original game it moves only left and right. Can be changed easily.

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
+let spaceship;
+let canvas;
+
 function setup(){
-  createCanvas(800,800);
+  canvas = createCanvas(document.body.clientWidth, 500);
```
Comment (Trion129) [inline js/index.js]: p5 has variable windowHeight and windowWidth, they can be used here

File: js/index.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
+let spaceship;
```
Comment (Trion129) [inline js/index.js]: Variables should use only var to support old browsers or the code should be transpiled using browserify.
Also all global variables should be in variables file

File: js/controlHandlers.js [hunk 3]
```diff
@@ -1,18 +1,36 @@
...
+    return code === 32;
+  }
 }
 
+
 //Call update functions of all game objects
-function updatePositions(){
-  //TODO
+function updatePositions() {
+  spaceship.update();
```
Comment (Trion129) [inline js/controlHandlers.js]: Iterate over all game objects and trigger their update functions instead of just spaceship

File: js/spaceShip.js [hunk 4]
```diff
@@ -1,33 +1,62 @@
...
+  this.draw = function () {
+    image(this.img, this.pos.x, this.pos.y, 100, 65);
   }
 }
 
 
 //Generate SpaceShip with constructor, give the x and y
-function placeSpaceShip(){
+function placeSpaceShip() {
+  const spaceShip = new SpaceShip(canvas.width / 2, canvas.height - canvas.height / 5);
```
Comment (Trion129) [inline js/spaceShip.js]: Use var please :P
Also push spaceship to gameObjects so its update and draw functions can be called

File: js/spaceShip.js [hunk 5]
```diff
@@ -1,33 +1,62 @@
-var SpaceShip = function(x, y){
+
+
+  //default parameters are to put the spaceShip in the center
+var SpaceShip = function (x = width / 2, y = height / 2) {
```
Comment (Trion129) [inline js/spaceShip.js]: Default values are again ES6 and not supported on some browsers, you can use an if condition in the function to set default values instead

File: js/spaceShip.js [hunk 6]
```diff
@@ -1,33 +1,62 @@
...
+    switch (moveCode) {
       //TODO make decisions here and apply Force towards that side in acc
-      // this.acc.add(forceVector);
+      // acc.add(forceVector);
+      case DOWN_ARROW:
+        forceVector.set(0,1);
+        break;
+      case UP_ARROW:
+        forceVector.set(0,-1);        
+        break;
```
Comment (Trion129) [inline js/spaceShip.js]: Up and down motions are just too OP, must be removed before a possible merge

**Review by Trion129 — CHANGES_REQUESTED** [review]
I made some reviews, tweak the code accordingly ☺👍

**Comment by chiptus** [comment]
Updated with the requested changes

File: js/controlHandlers.js [hunk 7]
```diff
@@ -26,7 +26,7 @@ function setupControls() {
 
 //Call update functions of all game objects
 function updatePositions() {
-  spaceship.update();
+  gameObjects.forEach(o => o.update());
```
Comment (Trion129) [inline js/controlHandlers.js]: ES6 Arrow functions :P convert to anonymous functions

File: js/controlHandlers.js [hunk 8]
```diff
@@ -1,18 +1,36 @@
+
+
 /*
 ** Bind Keys and events to the spaceShip's controllers
 ** e.g. if someone presses space, create a bullet object
 */
-function setupControls(){
-  //TODO
+function setupControls() {
+  $(document).keydown(({which:code}) => {
```
Comment (Trion129) [inline js/controlHandlers.js]: ES6 Arrow functions should be anonymous functions

File: js/controlHandlers.js [hunk 9]
```diff
@@ -1,18 +1,36 @@
...
+    return code === 32;
+  }
 }
 
+
 //Call update functions of all game objects
-function updatePositions(){
-  //TODO
+function updatePositions() {
+  gameObjects.forEach(o => o.update());
```
Comment (Trion129) [inline js/controlHandlers.js]: Arrow functions :P

**Review by Trion129 — CHANGES_REQUESTED** [review]
Reviewed more needed changes, minor ones now

File: js/controlHandlers.js [hunk 10]
```diff
@@ -1,18 +1,36 @@
...
+function updatePositions() {
+  gameObjects.forEach(function (e) { o.update(); });
 }
 
 
 //Call draw functions of all game objects
-function drawObjects(){
-  //TODO
+function drawObjects() {
+  spaceship.draw();
```
Comment (Trion129) [inline js/controlHandlers.js]: This should also iterate over all game objects and trigger their draw function

**Review by Trion129 — CHANGES_REQUESTED** [review]
Amazing work, just 1 more change :-)

File: js/controlHandlers.js [hunk 11]
```diff
@@ -1,18 +1,36 @@
...
+    return code === 32;
+  }
 }
 
+
 //Call update functions of all game objects
-function updatePositions(){
-  //TODO
+function updatePositions() {
+  gameObjects.forEach(function (e) { o.update(); });
```
Comment (Trion129) [inline js/controlHandlers.js]: You are passing e to the function but updating undefined variable o

File: js/controlHandlers.js [hunk 12]
```diff
@@ -1,18 +1,36 @@
...
 }
 
 
 //Call draw functions of all game objects
-function drawObjects(){
-  //TODO
+function drawObjects() {
+  gameObjects.forEach(function (e) { o.draw(); });
```
Comment (Trion129) [inline js/controlHandlers.js]: Same here

**Review by Trion129 — CHANGES_REQUESTED** [review]

**Review by Trion129 — APPROVED** [review]
Well Done :-)

**Pull request merged**